primitives/messages/src/source_chain.rs

```diff
@@ -135,6 +135,15 @@ pub trait MessageDeliveryAndDispatchPayment<AccountId, Balance> {
 	}
 }
 
+/// Handler for delivered messages.
+#[impl_trait_for_tuples::impl_for_tuples(30)]
+pub trait OnMessagesDelivered {
+	/// Called when we receive confirmation that our messages have been delivered to the
+	/// target chain. The confirmation aso has single bit dispatch result for every
+	/// confirmed message (see `DeliveredMessages` for details).
+	fn on_messages_delivered(_lane: &LaneId, _messages: &DeliveredMessages) {}
```

Comment:
This'll need to do something with weights - I'm not yet sure how to deal with this. Change benchmarks to worst-case result and then refund? But what is worst case when there are e.g. multiple apps using different lanes for their purposes. And because of refund policy we'll need to decrease max number of messages in the confirmation tx. Probably just reserve single read and single write for the every registered callback? I'm not yet sure - will leave it for future PRs as well

Reply:
Yeah, I think the general sentiment about these kind of handlers is that they should be as simple as possible. Having one read + one write seems reasonable. In future we can prepare some tooling to kind of "subscribe" to interesting messages and for instance have them processed in `on_initialize` of the pallet or some manual trigger (extrinsic).

Might be good to add this assumption to the docs though, so that implementers are aware of that.

Reply:
#1012